src/Servers/Kestrel/Transport.Quic/src/Internal/QuicConnectionListener.cs

```diff
@@ -37,9 +37,11 @@ public QuicConnectionListener(QuicTransportOptions options, IQuicTrace log, EndP
 
             quicListenerOptions.ServerAuthenticationOptions = sslServerAuthenticationOptions;
             quicListenerOptions.ListenEndPoint = endpoint as IPEndPoint;
-            quicListenerOptions.IdleTimeout = options.IdleTimeout;
             quicListenerOptions.MaxBidirectionalStreams = options.MaxBidirectionalStreamCount;
             quicListenerOptions.MaxUnidirectionalStreams = options.MaxUnidirectionalStreamCount;
+            // Set Quic idle timeout to a conservative 10 minutes - we handle idle itemouts gracefully in the Kestrel layer
+            // based off of KestrelServerLimits.KeepAliveTimeout
+            quicListenerOptions.IdleTimeout = TimeSpan.FromMinutes(10);
```

Comment:
KeepAliveTimeout isn't implemented in Kestrel layer for HTTP/3. We're using the QUIC idle timeout.

Reply:
Is there a reason not to hard-code this to Kestrel's default 130 second KeepAliveTimeout?

Reply:
It's not configurable so we might want to be more generous?

Reply:
Given the pushback I got increasing this default by 10 seconds, I'm surprised we'd want to increase it nearly five-fold for Quic while making it non-configurable. Why is making it more generous better than making it more conservative? I'd just keep the default.

Reply:
Why aren't we making it configurable? I thought the idea here was to make IdleTimeout = KestrelServerLimits.KeepAliveTimeout

Reply:
Stephen's argument earlier was:

> If we want to continue exposing QuicTransportOptions.IdleTimeout fine, but I don't think we have to because it is redundant with the KeepAliveTimeout and the shorter one would win.

Which I buy. Having the extra API surface in the form of the Feature doesn't seem fully necessary. As for default vs. 10 min, am I misunderstanding that the shorter one would win today? If that's the case it doesn't seem like it would matter, so maybe it's cleaner to have them use the same 130 second default.

Reply:
I think the better layering is to have Kestrel.Core track the idleness of the connection and close the connection manually after the KeepAliveTimeout like we do for non-multiplexed transports rather than require the transport fish out a feature passed to BindAsync and implement a KeepAliveTimeout/IdleTimeout itself. We figured doing this might be a bit too much work for .NET 6, but adding a new public API just to remove it in the next release also feels bad, so we ended up here. See https://github.com/dotnet/aspnetcore/pull/35202#discussion_r686134545.

Reply:
Spicy: Make QuicTransportOptions.IdleTimeout internal. Kestrel sets it to KeepAliveTimeout.

We're all in the shared framework.